### ISSUE TITLE
DataTable resize visual bugs

### ISSUE THREAD

**Issue by primperfum** [primefaces/primereact#2433]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://forum.primefaces.org/viewforum.php?f=57
```

**Current behavior**
When resizing dataTable columns bizarre things happen: 
1.  When stretching columns with 'expand' mode, when dataTable gets wider than at the beggining the content of columns isn't visible. 
![image](https://user-images.githubusercontent.com/94358969/141765872-c80f4c42-191f-448c-923c-baa2348075ce.png)
2. When shrinking columns with 'expand' mode, when dataTable gets smaller than initial width rows go beyond dataTable.
![image](https://user-images.githubusercontent.com/94358969/141766514-baa598ea-c180-44b8-bf32-ff3be47e875d.png)

**Expected behavior**

**Minimal reproduction of the problem with instructions**

**Please tell us about your environment:**

* **React version:**

* **PrimeReact version:**

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

**Comment by mertsincan** [comment]
Interesting! Could you please attach a codesandbox link for us to replicate? Thanks a lot for your report!

**Comment by primperfum** [comment]
Hey, sure! Big thanks for inspecting those issues so fast :)  
https://codesandbox.io/s/dt-resize-lst3u?file=/src/App.js
It seems like virtualScroller makes it worse.
Before Vscroll, theres something wrong with last column: 
![Zrzut ekranu z 2021-11-19 11-16-33](https://user-images.githubusercontent.com/94358969/142607009-36002ef5-b462-4023-ab23-503a7bf70a34.png)
After virtualScroll: 
![Zrzut ekranu z 2021-11-19 11-09-36](https://user-images.githubusercontent.com/94358969/142607144-0c7d481a-2cd6-43fa-8777-374c1ce49cde.png)
![Zrzut ekranu z 2021-11-19 11-09-50](https://user-images.githubusercontent.com/94358969/142607160-9a6858ab-ee8c-4cd1-8a5b-8951eeb26fd1.png)

**Comment by primperfum** [comment]
The problem seems to be that on Your onColumnResizeEnd method in Datatable, if the mode is expand You do: 
  else if (this.props.columnResizeMode === 'expand') {
                const tableWidth = this.table.offsetWidth + delta + 'px';
                this.table.style.width = tableWidth;
                this.table.style.minWidth = tableWidth;
                this.resizeTableCells(newColumnWidth);
            }
And if there is virtualScroll, its content width is less than the table -> this is why it doesn't expand, because the parent of the rows which have flex: '1 1 width' (to resize) still has the same, old width -> not equal to table width and the rows are based on the old width.

When I manually set document.querySelector('.p-virtualscroller') && document.querySelector('.p-virtualscroller-content') widths to match table width it works just fine.
PS: There is also the issue that columns from header aren't exactly aligned with columns from body after doing the above fix (maybe it's just my case). It seems to be caused by 'scrollable' and scroll appearing on dataTable -> when I set marginRight of the tableHeader to match scroll width the bug doesnt appear anymore.
